### PR TITLE
feat: GIDK-439 | Prateek | Dev | Provide credit note linking with invoice #5152

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 1]
```diff
@@ -906,13 +914,51 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
...
+        this.selectedInvoiceForCreditNote = null;
+    }
+
+    /**
+     * Fetches the invoice list data for a voucher
+     *
+     * @param {*} event Contains the selected voucher details
+     * @memberof NewLedgerEntryPanelComponent
+     */
+    public getInvoiceListsData(event: any) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts:954]: add return type

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 2]
```diff
@@ -1147,8 +1161,14 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
 
-    public getInvoiveListsData(e: any) {
-        if (e.value === 'rcpt') {
+    /**
+     * Fetches the invoice list data for a voucher
+     *
+     * @param {*} event Contains the selected voucher details
+     * @memberof UpdateLedgerEntryPanelComponent
+     */
+    public getInvoiceListsData(event: any) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1170]: return type

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 3]
```diff
@@ -1160,11 +1180,67 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
+            voucherType: VoucherTypeEnum.creditNote
+        }
+        let date;
+        if (typeof this.vm.selectedLedger.entryDate === 'string') {
+            date = this.vm.selectedLedger.entryDate;
+        } else {
+            date = moment(this.vm.selectedLedger.entryDate).format(GIDDH_DATE_FORMAT);
+        }
+        this.invoiceList = [];
+        this._ledgerService.getInvoiceListsForCreditNote(request, date).subscribe((response: any) => {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1208]: remove _  (underscore) for variable name

File: apps/web-giddh/src/app/ledger/ledger.vm.ts [hunk 4]
```diff
@@ -355,6 +355,15 @@ export class BlankLedgerVM {
     public selectedCurrencyToDisplay?: 0 | 1 = 0;
     public passportNumber?: string;
     public touristSchemeApplicable?: boolean;
+    public invoiceLinkingRequest?: IInvoiceLinkingRequest;
+}
+
+export class IInvoiceLinkingRequest {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/ledger.vm.ts:361]: Add Doc

**Pull request merged**